src/Core/Components/Checkbox/FluentCheckbox.razor.cs

```diff
@@ -0,0 +1,126 @@
+// ------------------------------------------------------------------------
+// MIT License - Copyright (c) Microsoft Corporation. All rights reserved.
+// ------------------------------------------------------------------------
+
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection.Metadata;
+using Microsoft.AspNetCore.Components;
+using Microsoft.FluentUI.AspNetCore.Components.Utilities;
+using Microsoft.JSInterop;
+
+namespace Microsoft.FluentUI.AspNetCore.Components;
+
+/// <summary>
+/// The FluentCheckbox component is used to render a checkbox input...
+/// Test WIP
+/// </summary>
+public partial class FluentCheckbox : FluentInputBase<bool?>, IFluentComponentElementBase
+{
+    private const string JAVASCRIPT_FILE = FluentJSModule.JAVASCRIPT_ROOT + "Checkbox/FluentCheckbox.razor.js";
+
+    /// <inheritdoc />
+    [Parameter]
+    public ElementReference Element { get; set; }
+
+    /// <summary>
+    /// Gets or sets the checked state of the checkbox
+    /// </summary>
+    [Parameter]
+    public bool? Checked { get; set; }
+
+    /// <summary>
+    /// Gets or sets the indeterminate state of the checkbox
+    /// </summary>
+    [Parameter]
+    public bool? Indeterminate { get; set; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the CheckBox will allow three check states rather than two.
+    /// </summary>
+    [Parameter]
+    public bool ThreeState { get; set; }
+
+    /// <summary>
+    /// Gets or sets a value indicating the order of the three states of the CheckBox.
+    /// False(by default), the order is Unchecked -> Checked -> Intermediate.
+    /// True: the order is Unchecked -> Intermediate -> Checked.
+    /// </summary>
+    [Parameter]
+    public bool ThreeStateChecked { get; set; }
+
+    /// <summary>
+    /// Gets or sets the shape of the checkbox
+    /// </summary>
+    [Parameter]
+    public CheckboxShape Shape { get; set; } = CheckboxShape.Square;
+
+    /// <summary>
+    /// Gets or sets the content to prefix the input component.
+    /// </summary>
+    [Parameter]
+    public virtual RenderFragment? StartTemplate { get; set; }
+
+    /// <summary>
+    /// Gets or sets the content to suffix the input component.
+    /// </summary>
+    [Parameter]
+    public virtual RenderFragment? EndTemplate { get; set; }
+
+    /// <summary>
+    /// Gets or sets the size of the checkbox. See <see cref="Components.CheckboxSize"/>
+    /// </summary>
+    [Parameter]
+    public CheckboxSize Size { get; set; } = CheckboxSize.Medium;
+
+    /// <inheritdoc />
+    protected override void OnParametersSet()
+    {
+        base.OnParametersSet();
+        if (HasLabel())
+        {
+            // When the id is not provided, generate a unique id. This allow to use the label for.
+            Id ??= $"checkbox-{Identifier.NewId()}";
```

Comment:
It's not necessary to add a prefix. You can use `Identifier.NewId()`.
If the developer wants to know where his component is, he can set a custom ID.